Source/WebCore/css/calc/CSSCalcOperationNode.cpp

```diff
@@ -1369,6 +1370,14 @@ double CSSCalcOperationNode::evaluateOperator(CalcOperator op, const Vector<doub
     case CalcOperator::Tan: {
         if (children.size() != 1)
             return std::numeric_limits<double>::quiet_NaN();
+        double x = std::fmod(children[0], piDouble * 2);
```

Comment:
I’m not convinced that these special cases are sufficient. Perhaps we need even more. Does `tan(180deg)` correctly return `0`? I don’t see test coverage for that and I am concerned that it does not work correctly for similar reasons to these infinity cases. Separately, perhaps the special cases should be based on the return value of `std::tan` rather than on the arguments.

Reply:
I added testcases at: https://github.com/web-platform-tests/wpt/pull/40934